_posts/2022-05-31-introduction-to-project-panama-part-1.markdown

```diff
@@ -67,31 +68,30 @@ public static Linker getSystemLinker() {
 }
 ```
 
-In the JDK terminology, **Linker** is an instance of the platform-specific C ABI implementation.
-Linker offers a set of API methods designed to perform both down- and upcalls.
+In the JDK terminology, [**Linker**]((https://download.java.net/java/early_access/jdk19/docs/api/java.base/java/lang/foreign/Linker.html)) is an instance of the platform-specific C ABI implementation.
+Linker offers a set of API methods designed to perform both downcalls and upcalls, where:
+* a _downcall_ is an event initiated from a high-level subsystem,
+  in our case the JVM to a lower-level subsystem, like the OS kernel. I'll explain it later in the text in the context of Foreign Function & Memory API.
+* an _upcall_. The opposite of a downcall.
 
-While the **Linker** is like your cellphone - call whoever you want to, just dial in a proper phone number.
+While the **Linker** is like your telephone - call whoever you want to, just dial in a proper phone number.
 The symbol lookup methods are like your phonebook - just provide a proper descriptor of who you want to call!
 
 To perform a downcall, you need a descriptor of a function you’d like to call, a native address allocated through a symbol lookup,
 and a linker to create a method handle that you can invoke.
 
-To make it clear, let's briefly look at what both down- and up-calls are:
-* Downcall. [Wiki](https://en.wiktionary.org/wiki/downcall) says it's an event initiated from a high-level subsystem,
-  in our case the JVM to a lower-level subsystem, like the OS kernel. I'll explain it later in the text in the context of Foreign Function & Memory API.
-* Upcall. The opposite thing to a downcall.
-
-Let’s implement a classic C-style “hello world” but in Java:
+Implementing a classic C-style Hello World from Java:
 ```cpp
 int     printf(const char * __restrict, ...)
 ```
 
-## HelloWorld C-style in Java
+## "Hello World" C-style in Java
 
+This section will cover what has to be done in a form of tasks followed one by another in a logical sense.
```

Comment:
To write a Java-based Hello World that uses native `printf` functions, we need to